docfx_yaml/extension.py

```diff
@@ -396,11 +396,86 @@ def indent_code_left(lines, tab_space):
     return "\n".join(parts)
 
 
-def _parse_docstring_summary(summary):
+def _parse_enum_content(parts: Sequence[str]) -> Sequence[Mapping[str, str]]:
+    """Parses the given content for enums.
+
+    Args:
+        parts: The content to parse, given in the form of separated by
+            newlines and left-indented.
+
+    Returns:
+        Sequence of mapping of enum entries for name and description.
+
+    Raises:
+        ValueError: If the `Values` enum docstring is malformed.
+    """
+    enum_content: MutableSequence[Mapping[str, str]] = []
+    enum_name = ""
+    enum_description = []
+    prev_tab_space = -1
+    for part in parts:
+        if (
+            (current_tab_space := len(part) - len(part.lstrip(" "))) == 0 and
+            prev_tab_space != 0
+        ):
+            if prev_tab_space != -1 and (
+                (not enum_name or not enum_description)
+            ):
+                raise ValueError(
+                    "The enum docstring is not formatted well. Check the"
+                    " docstring:\n\n{}".format("\n".join(parts))
+                )
+            # Add the new enum and start collecting new entry.
+            new_entry = {
+                "id": enum_name,
+                "description": " ".join(enum_description),
+            }
+            if all(new_entry.values()):
+                enum_content.append(new_entry)
+            enum_description = []
+            # Only collect the name, not the value.
+            enum_name = part.split(" ")[0]
+        else:
```

Comment:
Nit: I would remove a bit of conditional nesting here by checking for this condition first. I think it would be `if enum_name and enum_description and current_tab_space != 0)`.

Reply:
Good idea. Done.